### PR TITLE
Eco 691 depreacted error

### PR DESCRIPTION
**Pull request by doodyparizada**

### CODE REVIEW

File: scripts/src/errors.ts [hunk 1]
```diff
@@ -291,3 +294,11 @@ export function TooManyRegistrations(message: string): MarketplaceError {
 export function TooMuchEarnOrdered(message: string): MarketplaceError {
 	return TooManyRequests(CODES.TooManyRequests.Amounts, message);
 }
+
+function DeprecationError(index: number, message: string): MarketplaceError {
+	return new MarketplaceError(410, index, "Request to a deprecated resource", message);
+}
+
+export function BlockchainEndpointChanged(message: string): MarketplaceError {
+	return TooManyRequests(CODES.Deprecated.BlockchainEndpointChanged, message);
```
Comment (YohayBar) [inline scripts/src/errors.ts]: TooManyRequests  should be DeprecationError?

File: scripts/src/errors.ts [hunk 2]
```diff
@@ -291,3 +294,11 @@ export function TooManyRegistrations(message: string): MarketplaceError {
 export function TooMuchEarnOrdered(message: string): MarketplaceError {
 	return TooManyRequests(CODES.TooManyRequests.Amounts, message);
 }
+
+function DeprecationError(index: number, message: string): MarketplaceError {
+	return new MarketplaceError(410, index, "Request to a deprecated resource", message);
```
Comment (YohayBar) [inline scripts/src/errors.ts]: why are we using 410 and not costs?

File: scripts/src/public/middleware.ts [hunk 3]
```diff
@@ -3,13 +3,23 @@ import * as bearerToken from "express-bearer-token";
...
+	next();
+} as express.RequestHandler;
 
 export function init(app: express.Express) {
 	app.use(httpContext.middleware as express.RequestHandler);
 	app.use(bearerToken());
 	app.use(requestLogger);
 	app.use(logRequest);
 	app.use(reportMetrics);
+	app.use(deprecationError);
```
Comment (YohayBar) [inline scripts/src/public/middleware.ts]: will there be an overhead of checking every request for this specific header?

File: scripts/src/errors.ts (hunk 2 above)
Comment (doodyparizada) [reply scripts/src/errors.ts]: `consts`? This is the same throughout the file. I don't mind changing it

File: scripts/src/errors.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/errors.ts]: yep. I haven't added a test yet, I opened the PR just for it to be triggered by travis. But this should be caught by a unit test

File: scripts/src/public/middleware.ts (hunk 3 above)
Comment (doodyparizada) [reply scripts/src/public/middleware.ts]: It's just another if, and I can also remove it on production for now. The client just needs a way to test the error functionality

File: scripts/src/errors.ts [hunk 4]
```diff
@@ -98,177 +130,177 @@ export class MarketplaceError extends Error {
 }
 
 function UnauthorizedError(index: number, message: string) {
-	return new MarketplaceError(401, index, "Unauthorized Request", message);
+	return new MarketplaceError(CODES.Unauthorized.code, index, "Unauthorized Request", message);
 }
 
 export function MissingToken() {
-	return UnauthorizedError(CODES.Unauthorized.MissingToken, "Request missing token");
+	return UnauthorizedError(CODES.Unauthorized.types.MissingToken, "Request missing token");
```
Comment (nitzantomer) [inline scripts/src/errors.ts]: this is becoming too verbose, how about:
```
function UnauthorizedError(key: keyof typeof CODES.Unauthorized.types, message: string) {
	return new MarketplaceError(CODES.Unauthorized.code,, CODES.Unauthorized.types[key], "Unauthorized Request", message);
}
```

And then:
```
export function MissingToken() {
    return UnauthorizedError("MissingToken", "Request missing token");
}
```

File: scripts/src/public/middleware.ts (hunk 3 above)
Comment (nitzantomer) [reply scripts/src/public/middleware.ts]: this is weird. not that we should care about the extra function for each request, it's just that this is needed for a very specific thing only on dev/test.
why can't we just open a specific endpoint for it?

**Review by nitzantomer — APPROVED** [review]

**Pull request merged**